### PR TITLE
update go version to 1.21

### PR DESCRIPTION
**Pull request by hgarvison**
Update the go version to 1.21 and update all dependencies

Passing pipelines:
Attestation: https://github.com/microsoft/confidential-aci-examples/actions/runs/7291866269
SKR: https://github.com/microsoft/confidential-aci-examples/actions/runs/7291996852
ENCFS: https://github.com/microsoft/confidential-aci-examples/actions/runs/7291997649

### CODE REVIEW

File: docker/attestation-container/Dockerfile [hunk 1]
```diff
@@ -8,15 +8,18 @@
...
+# https://github.com/microsoft/go/tree/microsoft/main/eng/doc/fips#usage-build
+ENV GOOS=linux GOEXPERIMENT=opensslcrypto CGO_ENABLED=1
+
 # pre-copy/cache go.mod for pre-downloading dependencies and only redownloading them in subsequent builds if they change
 COPY . .
 RUN go mod download && go mod verify
 
-# Note: Use CGO_ENABLED=0 to build statically-linked binaries deployable in scratch image
-RUN CGO_ENABLED=0 go build -v -o /usr/local/bin/app ./cmd/attestation-container/
+RUN go build -v -o /usr/local/bin/app ./cmd/attestation-container/
```
Comment (BryceDFisher) [inline docker/attestation-container/Dockerfile:23]: Is there a need to remove CGO_ENABLED=0?

Comment (BryceDFisher) [reply docker/attestation-container/Dockerfile:23]: Ah, you're trying to enable fips via the systemcrypto flag.

File: docker/attestation-container/Dockerfile [hunk 2]
```diff
@@ -8,15 +8,18 @@
 ARG variant=run
 
 # Build
-FROM mcr.microsoft.com/oss/go/microsoft/golang:1.19-fips-cbl-mariner2.0 AS builder-app
+FROM mcr.microsoft.com/oss/go/microsoft/golang:1.21-fips-cbl-mariner2.0 AS builder-app
 WORKDIR /usr/src/app
 
+# Set environment variables for golang fips build
+# https://github.com/microsoft/go/tree/microsoft/main/eng/doc/fips#usage-build
+ENV GOOS=linux GOEXPERIMENT=opensslcrypto CGO_ENABLED=1
```
Comment (BryceDFisher) [inline docker/attestation-container/Dockerfile]: Might as well set this to the recommended GOEXPERIMENT=systemcrypto instead of statically picking.

**Review by SethHollandsworth — APPROVED** [review]
lgtm

**Pull request merged**